pinot-plugins/pinot-file-system/pinot-s3/src/main/java/org/apache/pinot/plugin/filesystem/S3PinotFS.java

```diff
@@ -642,6 +654,12 @@ private CopyObjectRequest generateCopyObjectRequest(String copySource, URI dest,
       if (_ssekmsEncryptionContext != null) {
         copyReqBuilder.ssekmsEncryptionContext(_ssekmsEncryptionContext);
       }
+      if (_sseCustomerKey != null) {
+        copyReqBuilder
+            .sseCustomerAlgorithm(_sseCustomerAlgorithm)
+            .sseCustomerKey(_sseCustomerKey)
+            .sseCustomerKeyMD5(_sseCustomerKeyMD5);
+      }
     }
     return copyReqBuilder.build();
   }
```

Comment:
Not in the scope of this functionality, but in the `open` method the caught exception is imminently re-thrown. The method could be simplified by removing `try-catch`